pwiz_tools/Skyline/Model/Lib/BlibData/BlibDb.cs

```diff
@@ -136,6 +136,90 @@ public int GetSpectraCount()
             }
         }
 
+        private class ProteinTablesBuilder
+        {
+            private ISession _session;
+            private Dictionary<string, int> _namedProteinIds;
+            private Dictionary<int, int> _peptideIdProteinId;
+
+            public ProteinTablesBuilder(ISession session)
+            {
+                _session = session;
+                _namedProteinIds = new Dictionary<string, int>();
+                _peptideIdProteinId = new Dictionary<int, int>();
+            }
+
+            public void Add(DbRefSpectra refSpectra, string proteinName)
+            {
+                if (!string.IsNullOrEmpty(proteinName))
+                {
+                    if (!_namedProteinIds.TryGetValue(proteinName, out var proteinTableId))
+                    {
+                        proteinTableId = _namedProteinIds.Count + 1;
+                        _namedProteinIds.Add(proteinName, proteinTableId);
+                    }
+                    _peptideIdProteinId.Add((int)(refSpectra.Id ?? 0), proteinTableId);
+                }
+            }
+
+            public void Write()
+            {
+                // Output the protein - peptide relationships
+                using (var cmd = _session.Connection.CreateCommand())
+                {
+                    cmd.CommandText = @"DROP TABLE IF EXISTS Proteins";
+                    cmd.ExecuteNonQuery();
+                }
+                using (var cmd = _session.Connection.CreateCommand())
+                {
+                    cmd.CommandText = @"DROP TABLE IF EXISTS RefSpectraProteins";
+                    cmd.ExecuteNonQuery();
+                }
+
+                if (_namedProteinIds.Any())
+                {
+                    using (var cmd = _session.Connection.CreateCommand())
+                    {
+                        cmd.CommandText = @"CREATE TABLE Proteins (id INTEGER not null, accession TEXT)";

```

Comment:
Make sure you say that "id" is the primary key.
I recommend:
CREATE TABLE Proteins (id INTEGER primary key autoincrement, accession TEXT)

Reply:
Yup.